cli/src/cli_util.rs

```diff
@@ -2041,6 +2099,7 @@ See https://jj-vcs.github.io/jj/latest/working-copy/#stale-working-copy \
         ui: &Ui,
         mut tx: Transaction,
         description: impl Into<String>,
+        _git_head_lock: &GitHeadLock,
     ) -> Result<(), CommandError> {
         if !tx.repo().has_changes() {
```

Comment:
Locking isn't needed if the transaction has no changes. Perhaps, this check can be moved to caller.